examples/beautiful_mnist.py

```diff
@@ -23,12 +23,12 @@ def __call__(self, x:Tensor) -> Tensor: return x.sequential(self.layers)
   model = Model()
   opt = nn.optim.Adam(nn.state.get_parameters(model))
 
-  # TODO: there's a compiler error if you comment out TinyJit since randint isn't being realized and there's something weird with int
   @TinyJit
-  def train_step(samples:Tensor) -> Tensor:
+  def train_step() -> Tensor:
     with Tensor.train():
       opt.zero_grad()
       # TODO: this "gather" of samples is very slow. will be under 5s when this is fixed
+      samples = Tensor.randint(512, high=X_train.shape[0])
```

Comment:
Did you test this?

Reply:
Yes. Works on M2 Mac Pro.

Reply:
What accuracy did you get?

Reply:
Samples inside of train_step: 94.99% with `@TinyJit`, 98.16% without. 
Samples outside of train_step: 98.21% with `@TinyJit`, 98.24% without.

Jit is around 15% faster.

So something going on with the jit + random, will revert that change.